packages/application-generic/src/usecases/select-integration/select-integration.usecase.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+
+import {
+  GetNovuIntegration,
+  GetNovuIntegrationCommand,
+} from '../get-novu-integration';
+import { SelectIntegrationCommand } from './select-integration.command';
+import { decryptCredentials } from '../../encryption';
+
+@Injectable()
+export class SelectIntegration {
+  constructor(
+    private integrationRepository: IntegrationRepository,
+    private getNovuIntegration: GetNovuIntegration
+  ) {}
+
+  async execute(
+    command: SelectIntegrationCommand
+  ): Promise<IntegrationEntity | undefined> {
+    const query: Partial<IntegrationEntity> & { _organizationId: string } = {
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+      channel: command.channelType,
+      ...(command.providerId ? { providerId: command.providerId } : {}),
+      active: true,
+    };
+
+    const integration = await this.integrationRepository.findOne(
+      query,
+      undefined,
+      { query: { sort: { createdAt: -1 } } }
```

Comment:
Not in the requirements but as a team we might discuss if we want to favour to also sort them based in if they are active or not. Like:
First: newest and active
Second: newest and disabled
Third: oldest and disabled

Reply:
I'm not sure I do understand what you are saying. This use-case will be used in the send phase and it will look for the latest active integration, we might want to return in the future an array of active ones, but not sure why we want to have also inactive?

Reply:
Argh I mixed that because I thought was going to reuse this use case like the one we replaced (GetDecryptedIntegrations). Ignore me, please. 🙏🏻